### ISSUE TITLE
Can you explain the parameter configuration of using UA-DETRAC to train the CenterNet model？

### ISSUE THREAD

**Issue by asmallcat** [hu64/SpotNet#16]
I used the UA-DETRAC dataset to train the CenterNet model, but it did not achieve the performance in your paper. Can you explain your experimental process and parameter settings? thanks!

**Comment by asmallcat** [comment]
Which CenterNet are you using？CenterNet:Objects as points or CenterNet : Keypoint triplets for object detection? your paper show performance of CenterNet : Keypoint triplets for object detection is 83.48%, but I see your code use CenterNet:Objects as points.

**Comment by hu64** [comment]
Hi, we're definitely using objects as points, for the results as well, sorry for the confusion. For the hyperparameters, we are using --batch_size 2 and --lr 1e-5. We start the training with the pretrained CenterNet on COCO (available on the CenterNet GitHub repo). We train on 512x512 by evaluate with --keep_res. I hope this helps! We train by monitoring the performance on a small validation set every epoch. Once the validation performance no longer increases, we stop the training. For the evaluation, we used the official Matlab toolkit provided by the dataset authors.

**Comment by asmallcat** [comment]
thanks

**Comment by asmallcat** [comment]
Are you using a single GPU？

**Comment by hu64** [comment]
Yeah, a single GTX 1080 Ti

**Comment by asmallcat** [comment]
Hi，I am very sad that I  still cannot reproduce your UA-DETRAC results on CenterNet. Are your training set and test set randomly divided? I see that your data set is based on every 10 frames frames for training. How much is the difference between the results of every 10 frames frames and the results of training using all the data sets?thanks

**Comment by asmallcat** [comment]
![image](https://user-images.githubusercontent.com/22765966/137708365-dbede154-4aa7-46c7-9a5d-e5654144e85c.png)
this is val loss，it's abnormal.

**Comment by asmallcat** [comment]
and how many epochs you train ?

**Comment by hu64** [comment]
HI, I was training with a fixed train and val (from the train set), and I evaluated on the test set as defined by the website. Also yes, I trained with one image out of ten, although I don't think this really makes a difference on performance, it just saves time. As for your image, I don't think the curve is abnormal necessarily. It might be simply that you overtrain and should stop at epoch ~18. I am not sure how many epochs exactly, but one thing I can say is for evaluation, I was using the best model on the validation performance, "model_best" and not "model_last".

**Comment by asmallcat** [comment]
Hi, you trained with one image out of ten，and are the validation set and test set designed like this(one image out of ten)?There is a problem with the official website and I cannot submit the test results. But I use the evaluation index of the COCO data set, the reault of best_model in val mAP(IoU=0.75)=0.610, mAP(IoU=0.50)=0.639, which is very different from the performance of CenterNet in your paper.

**Comment by hu64** [comment]
Ah okay but in that case it’s kind of hard to compare. They have some custom ignore region rules for if a detection should count as false positive or not which makes a big difference in the evaluation… It really wouldn’t be fair to compare it with the coco evaluation in my opinion.

**Comment by asmallcat** [comment]
Thank you very much for your help. I cannot download the toolkit from the official website. Can you send me a copy of the evaluation tools for the UA-DETRAC test set?  I will evaluate the algorithms offline!  My email is blinkcz@163.com.

**Comment by hu64** [comment]
I just sent a link to that email address.
